libmcount/mcount.c

```diff
@@ -1716,6 +1724,145 @@ static void mcount_script_init(enum uftrace_pattern_type patt_type)
 	strv_free(&info.cmds);
 }
 
+/* unlink socket file if it exists */
+static void unlink_socket_maybe(struct sockaddr_un *addr)
+{
+	if (unlink(addr->sun_path) == -1) {
+		if (errno != ENOENT) {
+			pr_dbg("cannot unlink %s\n", addr->sun_path);
+		}
+	}
+}
+
+static int agent_init(struct sockaddr_un *addr)
+{
+	int sfd;
+
+	if (mkdir(MCOUNT_AGENT_SOCKET_DIR, 0775) == -1) {
+		if (errno != EEXIST) {
+			pr_dbg("error creating run directory %s\n", MCOUNT_AGENT_SOCKET_DIR);
+		}
+	}
+
+	sfd = socket_create(addr, getpid());
+	if (sfd == -1)
+		return sfd;
+
+	unlink_socket_maybe(addr);
```

Comment:
Why do you want to remove it?

Reply:
Socket binding fails when the file already exists. The only case I see it happening is if a PID is used twice by a uftrace target, and isn't cleared by the first one.

Reply:
Ok, then I think we should warn instead if the file exists.